src/sysusers/sysusers.c

```diff
@@ -392,11 +392,13 @@ static int write_files(void) {
 
                                 i = hashmap_get(groups, gr->gr_name);
                                 if (i && i->todo_group) {
+                                        log_error("error: group %s already exists", gr->gr_name);
                                         r = -EEXIST;
                                         goto finish;
                                 }
 
                                 if (hashmap_contains(todo_gids, GID_TO_PTR(gr->gr_gid))) {
+                                        log_error("error: group %s already in hashmap", gr->gr_name);
```

Comment:
The new message does not really help the user, IMO. The fact that we use a hashmap is an implementation detail the user should not need to care about. If this happens, we have a gid collision, and the message should state that. Also, the important information here is *which* gid clashed, so you should print that as well.